packages/app/ui/inputs/TextInput.tsx

```diff
@@ -0,0 +1,49 @@
+import { Input, InputProps, Field, Label } from "@headlessui/react";
+import { twMerge } from "tailwind-merge";
+
+import { BodyText, CaptionText } from "../text";
+
+interface TextInputxProps extends InputProps {
+  className?: string;
+  error?: string;
+  id: string;
+  label?: string;
+  wrapperClassName?: string;
+}
+
+export const TextInput = ({
+  className,
+  error,
+  id,
+  label,
+  onChange,
+  placeholder,
+  value,
+  wrapperClassName,
+  ...inputProps
+}: TextInputxProps) => (
+  <Field className={twMerge("flex flex-col gap-2", wrapperClassName)}>
+    {label && (
+      <Label htmlFor={id}>
+        <BodyText className="text-em-low">{label}</BodyText>
+      </Label>
+    )}
+    <Input
+      id={id}
+      value={value}
+      placeholder={placeholder}
+      onChange={onChange}
```

Comment:
Most of this props are already propagated on `inputProps`, right? We can not declare this if they aren't be used elsewhere.